From db7cf5ddd861a83bd81f5f91c5edc4728abb61e6 Mon Sep 17 00:00:00 2001
From: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date: Fri, 24 Jul 2026 23:10:02 +0200
Subject: [PATCH 03/63] UPSTREAM: hurd: Make the readlink __fstatat64
 references optional

They may show up or not depending on the toolchain in use.
---
 sysdeps/mach/hurd/i386/localplt.data   | 4 ++--
 sysdeps/mach/hurd/x86_64/localplt.data | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/sysdeps/mach/hurd/i386/localplt.data b/sysdeps/mach/hurd/i386/localplt.data
index 2befcd3748..5b9413422d 100644
--- a/sysdeps/mach/hurd/i386/localplt.data
+++ b/sysdeps/mach/hurd/i386/localplt.data
@@ -25,14 +25,14 @@ ld.so: __writev
 ld.so: __libc_lseek64
 ld.so: __mmap
 ld.so: __fstat64
-ld.so: __fstatat64
+ld.so: __fstatat64 ?
 ld.so: __stat64
 ld.so: __access
 ld.so: __getpid
 ld.so: __getcwd
 ld.so: _exit ?
 ld.so: abort
-ld.so: readlink
+ld.so: readlink ?
 ld.so: _hurd_intr_rpc_mach_msg
 ld.so: __errno_location
 ld.so: _dl_init_first
diff --git a/sysdeps/mach/hurd/x86_64/localplt.data b/sysdeps/mach/hurd/x86_64/localplt.data
index fda28cd983..cc39118d12 100644
--- a/sysdeps/mach/hurd/x86_64/localplt.data
+++ b/sysdeps/mach/hurd/x86_64/localplt.data
@@ -24,14 +24,14 @@ ld.so: __writev
 ld.so: __libc_lseek64
 ld.so: __mmap
 ld.so: __fstat64
-ld.so: __fstatat64
+ld.so: __fstatat64 ?
 ld.so: __stat64
 ld.so: __access
 ld.so: __getpid
 ld.so: __getcwd
 ld.so: _exit ?
 ld.so: abort
-ld.so: readlink
+ld.so: readlink ?
 ld.so: _hurd_intr_rpc_mach_msg
 ld.so: __errno_location
 ld.so: _dl_init_first
-- 
2.55.0

